Sources/Basic/FixableError.swift

```diff
@@ -0,0 +1,23 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright 2015 - 2016 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+public protocol FixableError: CustomStringConvertible {
```

Comment:
I think Basic is good, since we will want to support fixable errors from a variety of subsystems.  I like having a protocol for this, since (as others have said) we eventually want the fix-it information to be an form that an IDE can mechanically apply.